### DISCUSSION TITLE
Missing MDC information in http-access logs

### DISCUSSION THREAD

**Discussion by danielFesenmeyer** [keycloak/keycloak#23541] [Q&A]
We use the Keycloak JSON logs for our analytics system.

For almost all requests, we get a `mdc` object like that in the Keycloak JSON logs:
`{"trace_flags":"00","trace_id":"22b10a971f7ca7f8d37b4a1046a03953","span_id":"b82ec4436c356b4a"}`

We use the trace_id to find related logs (like errors that occurred during a certain request).

But for lots of http-access logs (logger `io.quarkus.http.access-log`), the mdc object is just an empty object: `{}`
It seems like the mdc object is filled for all other request-related logging.

I've observed that behavior with both Keycloak 20.0.5 and 22.0.3.

Does someone know why the MDC is sometimes available, and sometimes not?
Maybe it's related to RESTEasy Reactive https://github.com/keycloak/keycloak/issues/10713?

**Comment by sschu** [comment]
@danielFesenmeyer I agree this could be related. At least I remember that carrying diagnostic information has always been a problem in these reactive frameworks as they are typically stores in a ThreadLocal. @pedroigor WDYT?
